### PR TITLE
Fix Sorting entries in a taxonomy is broken

### LINKED ISSUE #2944: Taxonomies - Sorting entries in a taxonomy is broken
## Description
In taxonomies, if you edit a taxonomy and try to sort the entries ((they are called taxons internally)) with drag and drop, apparently the actions work but if you refresh the page, the order of the entries goes back to what it was before the change.

## Expected Behavior
The order set with drag and drop actions should remain.

## Actual Behavior
It's not possible to change the order of the entries in a taxonomy.

## Steps to Reproduce
Admin  > Configuration > Taxonomies and see that the sorting of taxonomies is broken.

## Context
https://community.openfoodnetwork.org/t/how-to-manage-product-categories-in-different-languages-in-a-multilingual-instances/1471/2

## Severity
S3? - a feature is broken and there is no workaround but the "sorting of product categories" is a very unimportant feature

## Your Environment
Local

* Version used: latest master
* Browser name and version: chrome 69
* Operating System and version (desktop or mobile): macos highsierra

## Possible Fix
I believe there's something in the management/updating of the spree_taxons.position DB field. I couldnt find the solution quickly... 
Meanwhile, If #2945 gets done, this issues would include reverting the alphabetical order and using the order defined by the user in the taxonomies screen.

### PR DESCRIPTION
**Pull request by macanudo527**
#### Fix Sorting entries in a taxonomy is broken

- Closes #2944 

The Coffeescript file `app/assets/javascripts/admin/spree/taxons/taxonomy.js.coffee` was using an older Taxon Controller, and so wasn't saving the order of taxons on the configuration menu. Updating it to use the `Spree::TaxonController` fixed the issue.

#### What should we test?

- Visit Admin > Configuration > Taxonomies page.
- Sort Taxons and refresh 

#### Release notes

The sorting of entries in a taxonomy is fixed.

Changelog Category: User facing changes

### CODE REVIEW

**Review by mkllnk — APPROVED** [review]
Thanks! :1st_place_medal: 

This is an excellent contribution. I just squashed your commits to make it easier for the next reviewer. We usually rewrite history before review and never during review but it's okay *between* reviews.

File: spec/controllers/spree/admin/taxons_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  context "as an admin" do
+    let(:current_api_user) { build(:admin_user) }
+
+    it "can reorder taxons" do
+      spree_post :update,
+                 taxonomy_id: taxonomy.id,
+                 id: taxon2.id,
+                 taxon: { parent_id: taxonomy.root.id, position: 0 }
+
+      expect(taxon2.reload.lft).to eq 2
```
[the remark this answers is not in the document]
Comment (macanudo527) [reply spec/controllers/spree/admin/taxons_controller_spec.rb:28]: Basically, the "taxon sorter" [here](https://github.com/openfoodfoundation/openfoodnetwork/blob/78b978efba1d4aecb3b6572f1603fb817064c041/app/assets/javascripts/admin/spree/taxons/taxonomy.js.coffee#L13-L17) sends a position and not `lft`, a complex formula [here](https://github.com/openfoodfoundation/openfoodnetwork/blob/78b978efba1d4aecb3b6572f1603fb817064c041/app/controllers/spree/admin/taxons_controller.rb#L45-L71) calculates `lft` and saves the record. This test just makes sure that conversion works properly. In theory, the order could be correct, but the `lft` could have an incorrect number. Or at least that was my thinking.

[the remark this answers is not in the document]
Comment (dacook) [reply spec/controllers/spree/admin/taxons_controller_spec.rb:28]: Ah, I see that is a complex formula! This `update` method therefore needs more tests around it to cover the cases for different tree positions. But hey we didn't have any tests before, so this is a big improvement!

**Review by dacook — APPROVED** [review]

**Comment by filipefurtad0** [comment]
Hey @neilnaveen ,

Thank you for addressing "oldie" issue!

**Before this PR**

Editing a taxonomy: `admin/taxonomies/<id>/edit`

Initial ordering:
image

After dragging-dropping:
image

After hitting Update:
image

So, we can see that, we can't really change the ordering.

**After this PR**
Repeating the steps above, we can verify it is possible to change the ordering of taxons:

image

We can see that everything still appears as before (alphabetical ordering) when editing a product, for example (outside of scope to change this behavior):

image

This looks great.
Thanks, merging.

**Pull request merged**